libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -1432,7 +1440,7 @@ substitutePred x tm = \case
   Let t b -> Let (substituteTerm [x := tm] t) (substituteBinder x tm b)
   ForAll t b -> ForAll (substituteTerm [x := tm] t) (substituteBinder x tm b)
   Case t bs -> Case (substituteTerm [x := tm] t) (mapList (substituteBinder x tm) bs)
-  Reify t f b -> Reify (substituteTerm [x := tm] t) f (substituteBinder x tm b)
+  Reifies t' t f -> Reifies (substituteTerm [x := tm] t') (substituteTerm [x := tm] t) f
```

Comment:
Again we hope the substituion makes it a Literal.

Reply:
Sorry, `simplifyTerm`.